misc/msvs/vcxproj.template

```diff
@@ -34,9 +35,31 @@
   <ItemGroup Condition="'$(IncludeListImported)'==''">
     %%DEFAULT_ITEMS%%
   </ItemGroup>
-  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" Condition=" '$(PlatformToolset)' != 'Clang' "/>
   <ImportGroup Label="ExtensionTargets">
   </ImportGroup>
+
+  <Target Name="Build" Condition=" '$(PlatformToolset)' == 'Clang' ">
+    <Exec Command="scons vsproj=yes dev_build=yes" Condition="'$(OS)' != 'Windows_NT'"/>
+    <Exec Command="scons use_mingw=yes use_llvm=yes vsproj=yes dev_build=yes" Condition="'$(OS)' == 'Windows_NT'"/>
+  </Target>
+  <Target Name="Rebuild" DependsOnTargets="Clean;Build" Condition=" '$(PlatformToolset)' == 'Clang' "/>
+  <Target Name="Clean" Condition=" '$(PlatformToolset)' == 'Clang' ">
+    <Exec Command="scons --clean vsproj=yes dev_build=yes" Condition="'$(OS)' != 'Windows_NT'"/>
+    <Exec Command="scons --clean use_mingw=yes use_llvm=yes vsproj=yes dev_build=yes" Condition="'$(OS)' == 'Windows_NT'"/>
+  </Target>
+
+  <ItemDefinitionGroup Condition=" '$(PlatformToolset)' == 'Clang' ">
+    <ClCompile>
+      <AdditionalIncludeDirectories>
+        %%INCLUDES%%;
+        %(AdditionalIncludeDirectories)
+      </AdditionalIncludeDirectories>
+      <PreprocessorDefinitions>
+        %%DEFINES%%;%(PreprocessorDefinitions)
+      </PreprocessorDefinitions>
+    </ClCompile>
+  </ItemDefinitionGroup>
```

Comment:
These should not be defined here, these are set in the per-build-target props file. The vcxproj should only be setting generic defines and properties that are not platform or build target specific, as much as possible. Any build-specific preprocessor definitions, includes, build arguments, anything like that needs to be set on the specific build target props file.

For example, in my scons-generated godot project, I have 2 different template target platforsm that use the clang toolset on windows. Any definition here that is set only by "is clang" will imply be shared by all of those platform builds, which will obviously break a lot of things.